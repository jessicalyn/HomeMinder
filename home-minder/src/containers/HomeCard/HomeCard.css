.home-card {
  background-color: white;
  border: solid 2px black;
  border-radius: 2rem;
  width: 75%;
  margin: auto;
  margin-top: 2rem;
  padding: 1.5rem
}

.home-form {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  width: 75%;
  margin: auto;
  align-items: center
}

label {
  padding-top: 1rem;
}

input {
  width: 60%;
  height: 1.5rem;
}

.card-button {
  width: 15rem;
  margin: 2rem;
  padding: 1rem;
  border-radius: 2rem;
  font-size: 1rem;
}