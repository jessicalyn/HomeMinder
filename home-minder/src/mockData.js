export const mockHomeData = [
  { id: "JK1234",
    room: "Kitchen",
    category: "Filters",
    name: "Refrigerator Water Filter",
    brand: "",
    model: "",
    lastReplaced: "",
    purchaseLocation: "",
    recommendation: "6 months",
    userScheduled: "",
    type: "Replace"
  },
  { id: "IF1234",
    room: "Kitchen",
    category: "Sponges",
    name: "Sponge",
    brand: "",
    model: "",
    lastReplaced: "",
    purchaseLocation: "",
    recommendation: "2-4 weeks",
    userScheduled: "",
    type: "Replace"
  },
  { id: "PF123",
    room: "Bathroom",
    category: "Toothbrush",
    name: "Toothbrush",
    brand: "",
    model: "",
    lastReplaced: "",
    purchaseLocation: "",
    recommendation: "1-3 months",
    userScheduled: "",
    type: "Replace"
  },
]