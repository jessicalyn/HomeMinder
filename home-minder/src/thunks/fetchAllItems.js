import { storeAllItems, hasError, isLoading } from '../actions'

export const fetchAllItems = (url) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const data = await response.json()
      dispatch(storeAllItems(data))
    } catch (error) {
      dispatch(hasError(error.message))
    }
  }
}