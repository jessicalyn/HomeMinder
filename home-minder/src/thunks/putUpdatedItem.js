import { isLoading, hasError, updateItem } from "../actions";

export const putUpdatedItem = (url, options) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, options)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      // dispatch(updateItem(result))
    } catch (error) {
      dispatch(hasError(error.message))
    }
  }
}