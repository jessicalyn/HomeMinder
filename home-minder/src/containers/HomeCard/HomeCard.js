import React, { Component } from 'react';
import { connect } from 'react-redux';
import './HomeCard.css'

export class HomeCard extends Component {
  constructor(){
    super();
    this.state = {
      brand: "",
      category: "",
      id: "",
      lastReplaced: "",
      model: "",
      name: "",
      purchaseLocation: "",
      replaceRecommend: "",
      room: "",
      type: "",
      userScheduled: "",
    }
  }

  updateValue = (e) => {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }

  storeUpdates = async (e) => {
    e.preventDefault()
    const data = this.state
    const url = `http://localhost:3001/api/v1/${this.state.id}`
    const options = {
      method: 'PUT',
      body: JSON.stringify(data),
      headers: { "Content-Type": "application/json" }
    }
    await this.props.putUpdatedItem(url, options)
    console.log("stored updates")
  }

  render() {
    const { lastReplaced, name, replaceRecommend, room, type } = this.props
    return(
      <div>
        <form className="audit-form" onSubmit={ this.storeUpdates }>
          <h2>{ name } in your { room }</h2>
          <h3>Complete the sections below to schedule your next Home Reminder</h3>
          <label>Brand: </label>
          <input type="text" name="brand" value={ this.state.brand } onChange={ this.updateValue }></input>
          <label>Model: </label>
          <input type="text" name="model" value={ this.state.model } onChange={ this.updateValue }></input>
          <label>Purchase Location: </label>
          <input type="text" name="purchaseLocation" value={ this.state.purchaseLocation } onChange={ this.updateValue }></input>
          <label>Last { type } Date: { lastReplaced } </label>
          <label>Recommended { type } Timeline: { replaceRecommend }</label>
          <label>Choose next { type } date:</label>
          <input type="text" name="userSchedule" value={ this.state.userScheduled } onChange={ this.updateValue }></input>
          <button>Add to Schedule</button>
        </form>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  items: state.items
})

export const mapDispatchToProps = (dispatch) => ({

})

export default connect(mapStateToProps, mapDispatchToProps)(HomeCard)