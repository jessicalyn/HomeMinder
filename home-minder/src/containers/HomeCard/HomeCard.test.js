import React from 'react';
import { shallow } from 'enzyme';
import { HomeCard } from './HomeCard';
import { mockHomeData } from '../../mockData';

import { putUpdatedItem } from '../../thunks/putUpdatedItem';
jest.mock('../../thunks/putUpdatedItem')

describe('HomeCard container', () => {
  describe('HomeCard', () => {
    let wrapper
    const mockItems = mockHomeData
    const mockputUpdatedItem = jest.fn()

    beforeEach(() => {
      wrapper = shallow(<HomeCard items={ mockItems } putUpdatedItem={ mockputUpdatedItem }/>)
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });

    it.skip('should have default state', () => {
      expect(wrapper.state('brand')).toEqual("")
      expect(wrapper.state('model')).toEqual("")
    });

    it.skip('should update the state on component did mount', () => {
      expect(wrapper.state('name')).toEqual("")
      wrapper.instance().componentDidMount()
      expect(wrapper.state('name')).toEqual("Refrigerator Water Filter")
    });

    it('should change state of brand when updateValue is invoked', () => {
      const mockBrandEvent = { target: { name: "brand", value: "Alexa" } }
      wrapper.instance().updateValue(mockBrandEvent)

      expect(wrapper.state('brand')).toEqual("Alexa")
    });

    it.skip('should change state of userScheduled when updateCalendar is invoked', () => {
      const mockDate = "Fri Apr 12 2019 00:00:00 GMT-0600 (Mountain Daylight Time)"

      expect(wrapper.state('userScheduled')).toEqual("")
      wrapper.instance().updateCalendar(mockDate)

      expect(wrapper.state('userScheduled')).toEqual("2019-04-12T06:00:00.000Z")
    });

    it('should invoke putUpdatedItem when handleSubmit is invoked', () => {
      wrapper.instance().handleSubmit()

      expect(mockputUpdatedItem).toHaveBeenCalled()
    });

    it.skip('should change the state of toHomepage to true when handleSubmit is invoked', () => {
      wrapper.setState({ toHomepage: false })

      wrapper.instance().handleSubmit()

      expect(wrapper.state('toHomepage')).toEqual(true)
    })
  });

  describe('mapStateToProps', () => {
    
  })
})