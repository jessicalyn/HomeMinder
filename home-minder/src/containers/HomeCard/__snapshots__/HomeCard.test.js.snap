// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCard container HomeCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeCard
    items={
      Array [
        Object {
          "brand": "",
          "category": "Filters",
          "id": "JK1234",
          "lastReplaced": "",
          "model": "",
          "name": "Refrigerator Water Filter",
          "purchaseLocation": "",
          "recommendation": "6 months",
          "room": "Kitchen",
          "type": "Replace",
          "userScheduled": "",
        },
        Object {
          "brand": "",
          "category": "Sponges",
          "id": "IF1234",
          "lastReplaced": "",
          "model": "",
          "name": "Sponge",
          "purchaseLocation": "",
          "recommendation": "2-4 weeks",
          "room": "Kitchen",
          "type": "Replace",
          "userScheduled": "",
        },
        Object {
          "brand": "",
          "category": "Toothbrush",
          "id": "PF123",
          "lastReplaced": "",
          "model": "",
          "name": "Toothbrush",
          "purchaseLocation": "",
          "recommendation": "1-3 months",
          "room": "Bathroom",
          "type": "Replace",
          "userScheduled": "",
        },
      ]
    }
    putUpdatedItem={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
           in your 
        </h2>,
        <h3>
          Complete the sections below to schedule your next Reminder
        </h3>,
        <form
          className="home-form"
          onSubmit={[Function]}
        >
          <label>
            Brand: 
          </label>
          <input
            name="brand"
            onChange={[Function]}
            type="text"
          />
          <label>
            Model: 
          </label>
          <input
            name="model"
            onChange={[Function]}
            type="text"
          />
          <label>
            Purchase Location: 
          </label>
          <input
            name="purchaseLocation"
            onChange={[Function]}
            type="text"
          />
          <label>
            Recommended 
             Timeline: 
            <span />
          </label>
          <label>
            Choose next 
             date:
          </label>
          <Calendar
            className="calendar"
            maxDetail="month"
            minDetail="century"
            name="userSchedule"
            onChange={[Function]}
            returnValue="start"
            showNavigation={true}
            showNeighboringMonth={true}
            view="month"
          />
          <button
            className="card-button"
          >
            Add to Schedule
          </button>
        </form>,
      ],
      "className": "home-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " in your ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " in your ",
          undefined,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Complete the sections below to schedule your next Reminder",
        },
        "ref": null,
        "rendered": "Complete the sections below to schedule your next Reminder",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Brand: 
            </label>,
            <input
              name="brand"
              onChange={[Function]}
              type="text"
            />,
            <label>
              Model: 
            </label>,
            <input
              name="model"
              onChange={[Function]}
              type="text"
            />,
            <label>
              Purchase Location: 
            </label>,
            <input
              name="purchaseLocation"
              onChange={[Function]}
              type="text"
            />,
            <label>
              Recommended 
               Timeline: 
              <span />
            </label>,
            <label>
              Choose next 
               date:
            </label>,
            <Calendar
              className="calendar"
              maxDetail="month"
              minDetail="century"
              name="userSchedule"
              onChange={[Function]}
              returnValue="start"
              showNavigation={true}
              showNeighboringMonth={true}
              view="month"
            />,
            <button
              className="card-button"
            >
              Add to Schedule
            </button>,
          ],
          "className": "home-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Brand: ",
            },
            "ref": null,
            "rendered": "Brand: ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "brand",
              "onChange": [Function],
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Model: ",
            },
            "ref": null,
            "rendered": "Model: ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "model",
              "onChange": [Function],
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Purchase Location: ",
            },
            "ref": null,
            "rendered": "Purchase Location: ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "purchaseLocation",
              "onChange": [Function],
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Recommended ",
                undefined,
                " Timeline: ",
                <span />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Recommended ",
              undefined,
              " Timeline: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Choose next ",
                undefined,
                " date:",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Choose next ",
              undefined,
              " date:",
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "calendar",
              "maxDetail": "month",
              "minDetail": "century",
              "name": "userSchedule",
              "onChange": [Function],
              "returnValue": "start",
              "showNavigation": true,
              "showNeighboringMonth": true,
              "view": "month",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add to Schedule",
              "className": "card-button",
            },
            "ref": null,
            "rendered": "Add to Schedule",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
             in your 
          </h2>,
          <h3>
            Complete the sections below to schedule your next Reminder
          </h3>,
          <form
            className="home-form"
            onSubmit={[Function]}
          >
            <label>
              Brand: 
            </label>
            <input
              name="brand"
              onChange={[Function]}
              type="text"
            />
            <label>
              Model: 
            </label>
            <input
              name="model"
              onChange={[Function]}
              type="text"
            />
            <label>
              Purchase Location: 
            </label>
            <input
              name="purchaseLocation"
              onChange={[Function]}
              type="text"
            />
            <label>
              Recommended 
               Timeline: 
              <span />
            </label>
            <label>
              Choose next 
               date:
            </label>
            <Calendar
              className="calendar"
              maxDetail="month"
              minDetail="century"
              name="userSchedule"
              onChange={[Function]}
              returnValue="start"
              showNavigation={true}
              showNeighboringMonth={true}
              view="month"
            />
            <button
              className="card-button"
            >
              Add to Schedule
            </button>
          </form>,
        ],
        "className": "home-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " in your ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " in your ",
            undefined,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Complete the sections below to schedule your next Reminder",
          },
          "ref": null,
          "rendered": "Complete the sections below to schedule your next Reminder",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Brand: 
              </label>,
              <input
                name="brand"
                onChange={[Function]}
                type="text"
              />,
              <label>
                Model: 
              </label>,
              <input
                name="model"
                onChange={[Function]}
                type="text"
              />,
              <label>
                Purchase Location: 
              </label>,
              <input
                name="purchaseLocation"
                onChange={[Function]}
                type="text"
              />,
              <label>
                Recommended 
                 Timeline: 
                <span />
              </label>,
              <label>
                Choose next 
                 date:
              </label>,
              <Calendar
                className="calendar"
                maxDetail="month"
                minDetail="century"
                name="userSchedule"
                onChange={[Function]}
                returnValue="start"
                showNavigation={true}
                showNeighboringMonth={true}
                view="month"
              />,
              <button
                className="card-button"
              >
                Add to Schedule
              </button>,
            ],
            "className": "home-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Brand: ",
              },
              "ref": null,
              "rendered": "Brand: ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "brand",
                "onChange": [Function],
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Model: ",
              },
              "ref": null,
              "rendered": "Model: ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "model",
                "onChange": [Function],
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Purchase Location: ",
              },
              "ref": null,
              "rendered": "Purchase Location: ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "purchaseLocation",
                "onChange": [Function],
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Recommended ",
                  undefined,
                  " Timeline: ",
                  <span />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Recommended ",
                undefined,
                " Timeline: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Choose next ",
                  undefined,
                  " date:",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Choose next ",
                undefined,
                " date:",
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "calendar",
                "maxDetail": "month",
                "minDetail": "century",
                "name": "userSchedule",
                "onChange": [Function],
                "returnValue": "start",
                "showNavigation": true,
                "showNeighboringMonth": true,
                "view": "month",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add to Schedule",
                "className": "card-button",
              },
              "ref": null,
              "rendered": "Add to Schedule",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
